rusoto/core/src/region.rs

```diff
@@ -249,18 +254,13 @@ impl Display for ParseRegionError {
     }
 }
 
-/// Get the region from `AWS_DEFAULT_REGION` environment variable.
-/// Uses us-east-1 if not set or value is malformed.
-pub fn default_region() -> Region {
-    match std::env::var("AWS_DEFAULT_REGION") {
-        Ok(v) => {
-            let slice : &str = &v;
-            match Region::from_str(slice) {
-                Ok(region) => region,
-                Err(_) => Region::UsEast1,
-            }
-        },
-        Err(_) => Region::UsEast1,
+impl Default for Region {
+    fn default() -> Region {
+        match std::env::var("AWS_DEFAULT_REGION") {
+            Ok(ref v) =>
```

Comment:
Check for empty String might be appropriate here as env can be set but empty.

Reply:
Just leaving my 2 cents.

Since Region::from_str(...) correctly handles incorrect strings I don't know if it makes sense. Adding in more branches isn't generally the best path.

Reply:
OK, then we should think about what behavior we want. Region::from_str(...) [will returns an error](https://github.com/rusoto/rusoto/blob/ea15dec7a16a45be22864a8b9676ffd02649d773/rusoto/core/src/region.rs#L233) if it receives an empty string. IMHO, we should not get an error if the  env variable is empty, but [return the default region](https://github.com/rusoto/rusoto/blob/ea15dec7a16a45be22864a8b9676ffd02649d773/rusoto/core/src/region.rs#L262) instead.

Reply:
I was wrong or unclear, or even worse both at the same time :-). If we (really) want to default to ```"us-east-1"``` then everything is alright. I am not yet sure how the other SDKs behave in this regard, whether they will raise an error or also default to ```"us-east-1"```.

Reply:
I agree it'd be worth looking into how other SDKs do it, however this is really just preserving existing behaviour in Rusoto so I'm less concerned about that for the moment! :)